src/Solvers/fourier_tridiagonal_poisson_solver.jl

```diff
@@ -25,7 +25,7 @@ architecture(solver::FourierTridiagonalPoissonSolver) = architecture(solver.grid
     D[N, m, n] = -1 / Δxᶠᵃᵃ(N, m, n, grid) - Δxᶜᵃᵃ(N, m, n, grid) * (λ1[m] + λ2[n])
 end 
 
-@kernel function compute_main_diagonals!(D, grid, λ1, λ2, dir::Val{2})
+@kernel function compute_main_diagonals!(D, grid, λ1, λ2, ::YDirection)
     m, n = @index(Global, NTuple)
     N = getindex(size(grid), 2)
```

Comment:
```suggestion
    i, k = @index(Global, NTuple)
    Ny = getindex(size(grid), 2)
```

is that right?

Reply:
In this PR (which precedes https://github.com/CliMA/Oceananigans.jl/pull/3117) I'm still using the notation `N`, `m`, `n`, `q` instead of `Nx`, `Ny`, `Nz`, `i`, `j`, `k`. But I'll change it later to what you suggested.